### PR TITLE
Add btc transmuter

### PR DESCRIPTION
**Pull request by Kukks**
Add Btc Transmuter to docker install

### CODE REVIEW

**Comment by NicolasDorier** [comment]
Why `BTCTRANSMUTER_CRYPTOS` ? you can't use what the user already set up with BTCPAYGEN_CRYPTOX ?

**Comment by Kukks** [comment]
Because the docker generator would need to become aware of the plugin to do
that. Did not want to pollute it with opt in services

**Comment by NicolasDorier** [comment]
The container does not close nicely.

**Comment by NicolasDorier** [comment]
What is this entrypoint ? 

https://github.com/btcpayserver/btcTransmuter/blob/master/BtcTransmuter/entrypoint.sh

Please either call dotnet process directly from dockerfile (like I did for NBXplorer) or use `exec dotnet ...` so that your process can see SIGKILL signals sent to it.

**Comment by NicolasDorier** [comment]
`NBXplorer is not configured from the looks of it.`

**Comment by NicolasDorier** [comment]
@Kukks need to limit the amount of stuff the user need to manually setup, else they will come in the chat and we lose time debugging them.

Maybe just build this parameter inside the `btcpay-setup.sh` from the BTCPAYGEN stuff.

**Comment by Kukks** [comment]
> What is this entrypoint ?
> 
> https://github.com/btcpayserver/btcTransmuter/blob/master/BtcTransmuter/entrypoint.sh
> 
> Please either call dotnet process directly from dockerfile (like I did for NBXplorer) or use `exec dotnet ...` so that your process can see SIGKILL signals sent to it.

The entrypoint is so that I can restart the app from inside the container itself. This is so I can reload runtime plugins if I ever add support.
Will remove for now

**Comment by Kukks** [comment]
> @Kukks need to limit the amount of stuff the user need to manually setup, else they will come in the chat and we lose time debugging them.
> 
> Maybe just build this parameter inside the `btcpay-setup.sh` from the BTCPAYGEN stuff.

I will create a generic env var that is built the generator that can be used

File: btcpay-setup.sh [hunk 1]
```diff
@@ -141,6 +140,17 @@ if [[ "$BTCPAYGEN_REVERSEPROXY" == "nginx" ]] && [[ "$BTCPAY_HOST" ]]; then
     BTCPAY_HOST="$DOMAIN_NAME"
 fi
 
+BTCPAY_CRYPTOS=""
+for i in $BTCPAYGEN_CRYPTO1 $BTCPAYGEN_CRYPTO2 $BTCPAYGEN_CRYPTO3 $BTCPAYGEN_CRYPTO4 $BTCPAYGEN_CRYPTO5 $BTCPAYGEN_CRYPTO5 $BTCPAYGEN_CRYPTO6 $BTCPAYGEN_CRYPTO7 $BTCPAYGEN_CRYPTO8
+do  
+    echo "checkign loop $i";
```
Comment (NicolasDorier) [inline btcpay-setup.sh]: typo

File: btcpay-setup.sh [hunk 2]
```diff
@@ -141,6 +140,17 @@ if [[ "$BTCPAYGEN_REVERSEPROXY" == "nginx" ]] && [[ "$BTCPAY_HOST" ]]; then
     BTCPAY_HOST="$DOMAIN_NAME"
 fi
 
+BTCPAY_CRYPTOS=""
+for i in $BTCPAYGEN_CRYPTO1 $BTCPAYGEN_CRYPTO2 $BTCPAYGEN_CRYPTO3 $BTCPAYGEN_CRYPTO4 $BTCPAYGEN_CRYPTO5 $BTCPAYGEN_CRYPTO5 $BTCPAYGEN_CRYPTO6 $BTCPAYGEN_CRYPTO7 $BTCPAYGEN_CRYPTO8
```
Comment (NicolasDorier) [inline btcpay-setup.sh]: should be in quote like `"$BTCPAYGEN_CRYPTO4"`

Comment (Kukks) [reply btcpay-setup.sh]: done

File: btcpay-setup.sh (hunk 1 above)
Comment (Kukks) [reply btcpay-setup.sh]: this was outdated

**Comment by NicolasDorier** [comment]
image

**Comment by Kukks** [comment]
Merge this when you can @NicolasDorier  :)

**Comment by NicolasDorier** [comment]
```
WARNING: The BTCPAY_CRYPTOS variable is not set. Defaulting to a blank string.
```

**Comment by NicolasDorier** [comment]
Oh maybe it is because I run btcpay-update without setup.

**Pull request merged**